contracts/templates/ERC20TemplatePredictoor.sol

```diff
@@ -0,0 +1,1155 @@
+pragma solidity 0.8.12;
```

Comment:
I'd like a better name than `ERC20TemplatePredictoor.sol`. (As we discussed).

Datapoint: We will likely have _more_ predictoor templates over time.
- Built-in oracle
- Different formula to compensate predictoors
- Bug fixes

Datapoint: `address.json` refers to templates by their index. 1 is ERC20Template, 2 is ERC20TemplateEnterprise, etc. As this list grows then it'll be harder to cross-reference. Ref:
```text
    "ERC20Template": {
      "1": "0x9b87E416a76d8bCAf59A5c746D5ceB57119F08Cb",
      "2": "0x12bB8D85a091A69A07E22E52d4567dBB91568f52",
      "3": "0x9497d1d64F2aFeBcd4f9916Eef3d9094E5Df962f"
    },
```

Datapoint: `ERC20TemplateEnterprise` is not a great name. Perhaps the original driver was an enterprise customer, but there are other potential users. The name has led to great confusion about where `ERC20TemplateEnterprise` is applicable.

[This comment](https://github.com/oceanprotocol/ocean.py/issues/1315#issuecomment-1432848930) gives further discussion and analysis of names. To go forward, it proposes a solution too: (a) have a _table_ that identifies the distinguishing characteristics of each template; and (b) from now on, label the templates ERC20Template3.sol, ERC20Template4.sol, etc.

Given the above - and as discussed - could we name _this_ new template "ERC20Template3.sol" please?

Reply:
Update: @alexcos20 created an issue to create this table

https://github.com/oceanprotocol/contracts/issues/754